lib/remote/apilistener-configsync.cpp

```diff
@@ -443,7 +443,16 @@ void ApiListener::SendRuntimeConfigObjects(const JsonRpcConnection::Ptr& aclient
 	Log(LogInformation, "ApiListener")
 		<< "Syncing runtime objects to endpoint '" << endpoint->GetName() << "'.";
 
-	for (const Type::Ptr& type : Type::GetAllTypes()) {
+	// Sync all objects in priority descending order, otherwise downtimes, comments... might be synced before their
+	// respective checkables are synchronized, which would result in comments/downtimes being ignored by the other
+	// endpoint since it does not yet know about their checkables. Since the runtime config updates event doesn't
+	// trigger a reload on the remote endpoint, these objects won't be synced again til the next reload.
+	std::vector<Type::Ptr> types = Type::GetAllTypes();
+	std::sort(types.begin(), types.end(), [](const Type::Ptr& a, const Type::Ptr& b) {
+		return a->GetActivationPriority() > b->GetActivationPriority();
```

Comment:
Better call Type#GetLoadDependencies().

Reply:
What is `Type#GetLoadDependencies()` supposed to bring me here? It would be nice if you could explain in more detail why you think activation priority is not right instead of simply suggesting "better call `Type#GetLoadDependencies()`".

Reply:
Activation priority is just about in which order objects start _doing things_. In contrast, Type#GetLoadDependencies() reflects the `load_after T;` we discussed offline. It fetches the types which must be _loaded before_ the given type.

Reply:
Theoretically we could even merge

* #9873
* #7786

instead.

Reply:
## Blocked by

* [ ] #10003

Reply:
@Al2Klimov Please explain those references to other PRs.

Does "could merge instead" mean that there already exist PRs that would fix the same issue?

Does "blocked by" mean you would want to use `Type::SortByLoadDependencies()` function added there instead of this `std::sort()` call?

Reply:
Exactly!

Reply:
> Theoretically we could even merge
> 
> * [Refresh #7786 #9873](https://github.com/Icinga/icinga2/pull/9873)
> * [WIP: Change the order of the API runtime object sync on reconnect #7786](https://github.com/Icinga/icinga2/pull/7786)
> 
> instead.

@yhabteab These two are still an option btw. 🙈